html/modules/custom/reliefweb_subscriptions/src/ReliefwebSubscriptionsMailer.php

```diff
@@ -0,0 +1,2142 @@
+    $subscription = $subscriptions[$sid];
+
+    if ($subscription['type'] === 'scheduled') {
+      $last = $this->getPreviousSendingTime($subscription);
+      $notification = [
+        'sid' => $sid,
+        'last' => $last,
+      ];
+      $data = $this->getScheduledNotificationData((object) $notification, $subscription);
+    }
+    else {
+      $payload = $subscription['payload'];
+      $payload['sort'] = ['id:desc'];
+      $payload['offset'] = 0;
+      $payload['limit'] = 1;
+
+      // Retrieve the data from the API.
+      $result = $this->query([], $subscription, $payload);
+      if (!empty($result) && !empty($result['data'][0]['fields'])) {
+        $data = $result['data'][0]['fields'];
+      }
+    }
+
+    // Get the mail subject.
+    $subject = $this->generateEmailSubject($subscription, $data);
+    if (empty($subject)) {
+      return '';
+    }
+
+    // Generate the HTML and text content.
+    $body = $this->generateEmailContent($subscription, $data);
+    if (empty($body)) {
+      return '';
+    }
+
+    // Render the email using the default frontend theme so that template
+    // overrides, if any, can be used.
+    $active_theme = $this->themeManager->getActiveTheme();
+    $default_theme_name = $this->themeHandler->getDefault();
+    $default_theme = $this->themeInitialization->getActiveThemeByName($default_theme_name);
+
+    $render_array = [
+      '#theme' => 'mimemail_message',
+      '#module' => 'reliefweb_subscriptions',
+      '#key' => 'notifications',
+      '#recipient' => '',
+      '#subject' => $subject,
+      '#body' => $body,
+
+    ];
+
+    $this->themeManager->setActiveTheme($default_theme);
+    $html = $this->renderer->renderRoot($render_array);
+    $this->themeManager->setActiveTheme($active_theme);
+
+    return [
+      'subject' => $subject,
+      'body' => $html,
+    ];
+  }
+
+  /**
+   * Trigger for sitreps.
+   */
+  protected function triggerOchaSitrepNotification($entity, $entity_type) {
+    if (!isset($entity->type) || $entity->type !== 'report') {
+      return FALSE;
+    }
+
+    $language = $entity->language ?? Language::LANGCODE_NOT_SPECIFIED;
+
+    // Check if the source is OCHA (id: 1503).
+    if (!empty($entity->field_source[$language])) {
+      $found = FALSE;
+      foreach ($entity->field_source[$language] as $item) {
+        // No strict equality as tid may be a numeric string.
+        if (isset($item['tid']) && $item['tid'] == 1503) {
+          $found = TRUE;
+          break;
+        }
+      }
+      if (!$found) {
+        return FALSE;
+      }
+    }
+
+    // Check if the format is Situation Report (id: 10).
+    if (!empty($entity->field_content_format[$language])) {
+      $found = FALSE;
+      foreach ($entity->field_content_format[$language] as $item) {
+        // No strict equality as tid may be a numeric string.
+        if (isset($item['tid']) && $item['tid'] == 10) {
+          $found = TRUE;
+          break;
+        }
+      }
```

Comment:
This is D7 code and doesn't work. Code should be replaced with things like `$entity->field_source->isEmpty()` and other FieldItemList methods.